agent/agent.go

```diff
@@ -110,11 +111,12 @@ func WithGitInfo(repository string, commitSha string, sourceRoot string) Option
 	}
 }
 
-func WithUserAgent(userAgent string) Option {
+func WithAgentType(agentType string) Option {
 	return func(agent *Agent) {
-		userAgent = strings.TrimSpace(userAgent)
-		if userAgent != "" {
-			agent.userAgent = fmt.Sprintf("%s %s", agent.userAgent, userAgent)
+		agentType = strings.TrimSpace(agentType)
+		if agentType != "" {
+			agent.agentType = agentType
+			agent.userAgent = fmt.Sprintf("%s %s", agent.userAgent, agentType)
```

Comment:
Let's just use `agentType` to replace the user agent instead of appending it

Reply:
`UserAgent` has the `{name}/{version}` format, so in this case the user agent `version` is gonna be missing. Is the backend with that?

Reply:
or maybe change the option to `WithAgentType(agentType, version string)` ?

Reply:
Then I think is best if we keep both `WithAgentType` and `WithUserAgent`, because we cannot build the user agent just with the agent type.

Reply:
Ok,

Reply:
Done, I kept the `WithUserAgent` but changed the behavior to replace instead of append